modules/png-extraction/ImageExtractor.py

```diff
@@ -102,19 +103,33 @@ def get_tuples(plan, outlist = None, key = ""):
         key =  key + "_"
     if not outlist:
         outlist = []
-    for aa  in plan.dir():
+    headers=[]
+    if(publicheadersonly):
+        for aa in plan.dir():
+            headers.append(plan[aa])
+    else:
+        feature_list = open("featureset.txt").read().splitlines()
+        if len(feature_list)==0:
+            headers=[i for i in plan]
+        else:
+            feature_list.append('PatientID')
+            feature_list.append('SeriesInstanceUID')
```

Comment:
Why are you hard-coding these 3 values here?


            feature_list.append('PatientID')
            feature_list.append('SeriesInstanceUID')
            feature_list.append('PhotometricInterpretation')
            
            
   The featureset.txt should be handled at a general level. Not just for private tags. If it exists, it can have both private and public fields. If it does not exist, you extract all the private and public fields.
   
   You can check https://github.com/Emory-HITI/Niffler/blob/dev/modules/meta-extraction/conf/featureset1.txt 

The last 6 lines are private headers. Meta-extraction already handles this.

Reply:
in image extraction function we are using these three tags. but if these tags are not specified in featureset.txt then these wont be extracted. and there will be an error

Reply:
Aha, you are smart!

Reply:
I think you should include StudyInstanceUID also. right?

Reply:
Thank you!

Reply:
can you please tell me where we are using it?

Reply:
line 185 and 195? https://github.com/Emory-HITI/Niffler/blob/dev/modules/png-extraction/ImageExtractor.py

Reply:
yes you are right! how shoud i edit it?
should i create a new pull request?
I am new to git so could you please help me?

Reply:
Yes, you can open a new pull request. We can close this.

Please make sure to test for true/false cases for this newly introduced attribute. I will test again tomorrow.

Reply:
okay. Thank you for the help.

Reply:
I have made a new commit. could you please check it? 
And I have tested this new attribute.

Reply:
I have opend a new pull request with all updates